Assets/MRTK/Core/Utilities/DebugUtilities.cs

```diff
@@ -30,27 +56,110 @@ public static void DebugAssert(bool condition)
         {
-            Debug.LogError(message);
+            LogError(message);
         }
 
         /// <summary>
         /// Logs a warning message.
         /// </summary>
         /// <param name="message">The message to log.</param>
+        [Obsolete("Deprecated. Use LogWarning instead.")]
         public static void DebugLogWarning(string message)
         {
-            Debug.LogWarning(message);
+            LogWarning(message);
         }
 
         /// <summary>
         /// Logs a message.
         /// </summary>
         /// <param name="message">The message to log.</param>
+        [Obsolete("Deprecated. Use Log instead.")]
         public static void DebugLog(string message)
         {
-            Debug.Log(message);
+            Log(message);
+        }
+
+        /// <summary>
+        /// Logs an error message.
+        /// </summary>
+        /// <param name="message">The message to log.</param>
+        public static void LogError(string message)
+        {
+            if (LogLevel <= LoggingLevel.Error)
+            {
+                Debug.LogError(message);
+            }
+        }
+
+        /// <summary>
+        /// Logs a warning message.
+        /// </summary>
+        /// <param name="message">The message to log.</param>
+        public static void LogWarning(string message)
+        {
+            if (LogLevel <= LoggingLevel.Warn)
+            {
+                Debug.LogWarning(message);
+            }
+        }
+
+        /// <summary>
+        /// Logs a message at the informational level.
+        /// </summary>
+        /// <param name="message">The message to log.</param>
+        public static void Log(string message)
+        {
+            if (LogLevel <= LoggingLevel.Information)
+            {
+                Debug.Log(message);
+            }
+        }
+
+        /// <summary>
+        /// Logs the given message to the Unity console and player log if verbose logging is enabled.
+        /// Note that verbose logs do not include the callstack
+        /// </summary>
+        /// <remarks>
+        /// If you are doing string concatentation or manipulation, use LogVerboseFormat
+        /// that formats the string, as that will not allocate memory when verbose logging is enabled.
+        ///
+        /// For example, don't do:
+        /// Debug.LogVerbose("This is my message: " + text);
+        ///
+        /// Do:
+        /// Debug.LogVerbose("This is my message: {0}", text);
+        /// </remarks>
+        public static void LogVerbose(string message)
+        {
+            if (LogLevel <= LoggingLevel.Verbose)
+            {
+                // Save/restore the previous stack trace configuration, rather than assuming that
+                // it's the default (StackTraceLogType.ScriptOnly)
+                StackTraceLogType stackTraceLogType = Application.GetStackTraceLogType(LogType.Log);
+                Application.SetStackTraceLogType(LogType.Log, StackTraceLogType.None);
```

Comment:
Based on comments in your docmentation

> ### Enable verbose logging
> 
> Indicates whether or not verbose MRTK logging will be enabled. This defaults to false, but can be
> turned on to take detailed traces that allow the MRTK team to debug/dig into issues.

 it seems like the StackTraceLogType should be set to `StackTraceLogType.Full` here and not StackTraceLogType.None?

Reply:
Ah, glad you left a comment on this.

I should have commented this up, but by default Debug.Log will log callstacks, which grows the unity player logs quite significantly. This makes it super hard to read the generated logs (because those call stacks are really really verbose). This is quite intentional to make parsing the logs easier, because otherwise they grow reallllly big